components/expenses/ExpenseForm.js

```diff
@@ -833,7 +834,13 @@ const ExpenseForm = ({
   const isDraft = expense?.status === expenseStatus.DRAFT;
   const [hasValidate, setValidate] = React.useState(validateOnChange && !isDraft);
   const intl = useIntl();
-  const initialValues = { ...getDefaultExpense(collective), ...expense };
+  const supportedExpenseTypes = React.useMemo(() => getSupportedExpenseTypes(collective), [collective]);
+  const isSingleSupportedExpenseType = supportedExpenseTypes.length === 1;
+  const initialValues = {
+    ...getDefaultExpense(collective),
+    type: !isDraft && isSingleSupportedExpenseType ? supportedExpenseTypes[0] : undefined,
```

Comment:
Looks like it would make sense to add this default type to `getDefaultExpense`, changing the function prototype to `getDefaultExpense(collective, supportedExpenseTypes)`, to make sure we always use the same default values regardless of the context.